data-products/src/main/scala/org/sunbird/analytics/sourcing/SourcingMetrics.scala

```diff
@@ -0,0 +1,173 @@
+package org.sunbird.analytics.sourcing
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.functions.{col, concat, count, lit, split}
+import org.apache.spark.sql.{DataFrame, Encoders, SQLContext, SparkSession}
+import org.ekstep.analytics.framework.Level.INFO
+import org.ekstep.analytics.framework.util.DatasetUtil.extensions
+import org.ekstep.analytics.framework.util.{HTTPClient, JSONUtils, JobLogger, RestUtil}
+import org.ekstep.analytics.framework.{FrameworkContext, IJob, JobConfig, Level, StorageConfig}
+import org.ekstep.analytics.model.ReportConfig
+import org.ekstep.analytics.util.Constants
+import org.sunbird.analytics.job.report.BaseReportsJob
+import org.sunbird.analytics.model.report.{TenantInfo, TenantResponse, TextbookData}
+import org.sunbird.analytics.util.TextBookUtils
+import org.sunbird.cloud.storage.conf.AppConf
+
+case class ContentHierarchy(identifier: String, hierarchy: String)
+case class TextbookReportResult(identifier: String, l1identifier: String,board: String, medium: String, grade: String, subject: String, name: String, chapters: String, channel: String, totalChapters: String)
+case class ContentReportResult(identifier: String, l1identifier: String, contentType: String)
+case class TextbookHierarchy(channel: String, board: String, identifier: String, medium: Object, gradeLevel: List[String], subject: Object,
+                             name: String, status: String, contentType: Option[String], leafNodesCount: Int, lastUpdatedOn: String,
+                             depth: Int, createdOn: String, children: Option[List[TextbookHierarchy]], index: Int, parent: String)
+case class FinalReport(identifier: String,l1identifier: String,board: String, medium: String, grade: String, subject: String, name: String, chapters: String, channel: String, totalChapters: String, slug:String)
+case class TextbookResponse(l1identifier:String,board: String, medium: String, grade: String, subject: String, name: String, chapters: String, channel: String)
+
+object SourcingMetrics extends optional.Application with IJob with BaseReportsJob {
+
+  implicit val className = "org.sunbird.analytics.job.SourcingMetrics"
+  val jobName: String = "Sourcing Metrics Job"
+  val sunbirdHierarchyStore: String = AppConf.getConfig("course.metrics.cassandra.sunbirdHierarchyStore")
```

Comment:
How do we get the cassandra ip from? Configure a cluster and let the cluster ip be input from the jobConfig

Reply:
The cassandra ip is being fetched from jobConfig variable itself in getReportingSparkContext(jobConfig)